packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -194,7 +195,10 @@ export function isFileServingAllowed(
 ): boolean {
   if (!server.config.server.fs.strict) return true
 
-  const file = fsPathFromUrl(url)
+  // When the base path is set, the safeModulesPath does not include the base prefix internally.
+  // Therefore, when retrieving a file from safeModulesPath, the base path should be stripped.
+  // See https://github.com/vitejs/vite/issues/9438#issuecomment-1465270409
+  const file = fsPathFromUrl(stripBase(url, server.config.rawBase))
```

Comment:
This function is called from `serveRawFsMiddleware`/`serveStaticMiddleware`/`transformMiddleware` that runs after `baseMiddleware` (this middleware strips the base). So I think we shouldn't run `stripBase` here.
https://github.com/vitejs/vite/blob/1292ad08a06e2ae8c44307797ba533f64a602759/packages/vite/src/node/server/index.ts#L614-L646

Reply:
So that's how it is. I remove stripBase here